optuna/study.py

```diff
@@ -322,6 +322,17 @@ def _should_stop() -> bool:
     def set_user_attr(self, key: str, value: Any) -> None:
         """Set a user attribute to the study.
 
+        Example:
+
+            .. testcode::
+
+                import optuna
+
+                study = optuna.create_study()
+
+                study.set_user_attr("contributors", ["Akiba", "Sano"])
+                study.set_user_attr("dataset", "MNIST")
```

Comment:
This comment is originally posted to #1745.
---

I think these values are taken from [this tutorial](https://optuna.readthedocs.io/en/stable/tutorial/attributes.html), It may be difficult for readers to imagine the actual usage. I guess they describe the objective function, but this example doesn't have objective functions. So, for example, how about adding a simple objective function and explaining it with the `user_attrs`?

```python
def objective(trial):
    x = trial.suggest_float("x", 0, 1)
    y = trial.suggest_float("y", 0, 1)
    return x ** 2 + y ** 2

...

study.set_user_attr("objective function", "quadratic fuction")
study.set_user_attr("dimensions", 2)
study.set_user_attr("contributors", ["Akiba", "Sano"])
```

What do you think of it? @HideakiImamura @norihitoishida

Reply:
I agree with you. Thank you!